observation_portal/common/schema.py

```diff
@@ -1,23 +1,15 @@
+from importlib import metadata
 from rest_framework import serializers
 from rest_framework.schemas.openapi import AutoSchema, SchemaGenerator
 from rest_framework.schemas.utils import is_list_view
 from rest_framework.viewsets import GenericViewSet
-from setuptools_scm import get_version
-from setuptools_scm.version import ScmVersion
-
-
-def version_scheme(version: ScmVersion) -> str:
-    """Simply return the string representation of the version object tag, which is the latest git tag.
-    setuptools_scm does not provide a simple semantic versioning format without trying to guess the next release, or adding some metadata to the version.
-    """
-    return str(version.tag)
 
 
 class ObservationPortalSchemaGenerator(SchemaGenerator):
     def get_schema(self, *args, **kwargs):
         schema = super().get_schema(*args, **kwargs)
         schema['info']['title'] = 'Observation Portal'
-        schema['info']['version'] = get_version(version_scheme=version_scheme, local_scheme='no-local-version')
+        schema['info']['version'] = metadata.version("django-ocs-observation-portal")
```

Comment:
I remember you alluding to this - this is how we get the auto-generated docs to display the correct version based on the git tag. Does this still allow for that, or do we need to somehow inject that into the doc build GitHub workflow?

https://github.com/observatorycontrolsystem/observation-portal/blob/main/.github/workflows/docs-gh-pages.yaml

Reply:
I'll test the doc generation locally to make sure, but I don't think we need any changes in that GH action.

The metadata version is coming from whatever is defined in `pyproject.toml`. That should be managed with `poetry version`. For example, to bump the patch version you'd do `poetry version patch`. Similar shortcuts exist for path, major, minor, prerelease, etc.

To actually create a release/tag, the workflow might have to change a little. You'd bump the version first and then commit and tag. This way the version number is checked into source code and is in that tag. E.g:
```
$ poetry version premajor # bump the version
$ git add pyproject.toml && git commit -m "version $(poetry version --short) # commit the version bump
$ git tag $(poetry version --short)
$ git push --tags
```
I can add this to the docs.

But if this is too cumbersome, I can look into other options too.

Reply:
I think that's fine, it's just a new workflow we'll have to get used to. Like anything else, document it and we'll try and make sure we adhere to it.